src/metabase/models/setting.clj

```diff
@@ -827,15 +827,15 @@
 
 (defn- audit-setting-change!
   [{:keys [name audit sensitive?]} previous-value new-value]
-  (let [maybe-obfuscate #(if sensitive? (obfuscate-value %) %)]
-   (audit-log/record-event!
-    :setting-update
-    {:details (merge {:key name}
-                     (when (not= audit :no-value)
-                       {:previous-value (maybe-obfuscate previous-value)
-                        :new-value      (maybe-obfuscate new-value)}))
-     :user-id api/*current-user-id*
-     :model  :model/Setting})))
+  (let [maybe-obfuscate #(cond-> % sensitive? obfuscate-value)]
+    (events/publish-event!
```

Comment:
Changed this to avoid a circular dependency between setting, audit-log, and premium-features.